Shared/Transforms/Transformation.cs

```diff
@@ -108,38 +118,62 @@ public partial class Transformation : Core.ICloneable
         /// Creates transformation object from results of <seealso cref="Actions.GetTransformResult"/>.
         /// </summary>
         /// <param name="dictionary">One can use <seealso cref="Actions.GetTransformResult.Info"/> array.</param>
-        public Transformation(Dictionary<string, object>[] dictionary) {
-            for (int i = 0; i < dictionary.Length; i++) {
-                if (i == dictionary.Length - 1) {
+        public Transformation(Dictionary<string, object>[] dictionary)
+        {
+            for (int i = 0; i < dictionary.Length; i++)
+            {
+                if (i == dictionary.Length - 1)
+                {
                     m_transformParams = dictionary[i];
                 }
-                else {
+                else
+                {
                     m_nestedTransforms.Add(new Transformation(dictionary[i]));
                 }
             }
         }
 
-        public Dictionary<string, object> Params {
+        public Dictionary<string, object> Params
+        {
             get { return m_transformParams; }
         }
 
-        public List<Transformation> NestedTransforms {
+        public List<Transformation> NestedTransforms
+        {
             get { return m_nestedTransforms; }
         }
 
         public bool HiDpi { get; private set; }
         public bool IsResponsive { get; private set; }
 
-        public Transformation Chain() {
+        public Transformation Chain()
+        {
+            Transformation nested = this.Clone();
+            nested.m_nestedTransforms = null;
+            m_nestedTransforms.Add(nested);
+            m_transformParams = new Dictionary<string, object>();
+            Transformation transform = new Transformation(m_nestedTransforms);
+            return transform;
+        }
+
+        protected void AssignVariable(Variable var)
```

Comment:
why protected?

Reply:
Because this is the internal method of the "Transformation" class. This should be hidden from users. However, child classes of the "Transformation" class should have access to this method.

Reply:
OK